consensus/src/consensus_observer/observer/execution_pool.rs

```diff
@@ -62,3 +69,545 @@ impl ObservedOrderedBlock {
         }
     }
 }
+
+/// Returns all ordered blocks for the given ordered block with window.
+/// This requires traversing backward via the parent links to identify
+/// and fetch the blocks from the pending block store. The blocks are
+/// returned in chronological order, and if any block is missing, this
+/// will return None.
+pub fn get_all_blocks_for_window(
+    pending_block_store: Arc<Mutex<PendingBlockStore>>,
+    ordered_block: &OrderedBlock,
+    window_size: u64,
+) -> Option<Vec<OrderedBlock>> {
```

Comment:
Should it be a `Result<_>` instead of an `Option<_>`?

Reply:
Done :)